docs/user_guide/advanced/steam_handler_module.md

```diff
@@ -0,0 +1,284 @@
+# Data Stream Handler module
+
+This section describes the architecture, usage, and available implementations of data stream handlers within CoFMPy.
+
+---
+## 1. Overview
+
+The `data_stream_handler` module provides a unified interface for ingesting and interacting with live or recorded data streams, which serve as inputs to Digital Twin prototypes. It enables users to seamlessly switch between data sources (e.g., CSV files, Kafka topics, or local in-memory streams) through a consistent API. This abstraction layer allows developers to prototype and test Digital Twins without rewriting data ingestion logic.
+
+### Core Concepts
+
+* **Modularity & Extensibility:** All stream handlers inherit from `BaseDataStreamHandler`, which defines the standard interface and registration mechanism for new handlers.
+* **Handler Factory:** Stream handler instances are dynamically instantiated using the `BaseDataStreamHandler.create_handler()` factory method, based on the configuration dictionary.
+* **Grouped data streams:** Control over dynamic instantiation and variable assignment is provided by `BaseDataStreamHandler.is_equivalent_stream()` and `BaseDataStreamHandler.add_variable()` methods, respectively.
+* **Alias Mapping:** A mechanism for mapping CoFMPy internal connections (as `(node, endpoint)` tuples) to external variables in the input data stream.
+* **Data Interpolation:** Each handler supports time-based interpolation (e.g., step-wise or linear), using the shared `Interpolator` utility.
+
+### Included Handlers
+
+* **CSVDataStreamHandler** (`csv`): Reads data from a static CSV file.
+* **KafkaDataStreamHandler** (`kafka`): Connects to a Kafka topic, consumes real-time messages in a separate thread, and manages buffering and interpolation.
```

Comment:
All stream handlers manage interpolation. Maybe, it could be removed here, since it is already mentioned in line 16.